### ISSUE TITLE
Postcss-cssnext and precss breaking all plugins

### ISSUE THREAD

**Issue by Skilgarriff** [postcss/postcss-loader#138]
I'm currently experiencing this weird bug that seems to only occur with the Postcss-cssnext and precss plugins.

Forgot to mention my versions:
Webpack: 2.1.0-beta.25
Postcss-Loader:  1.1.1
Precss: 1.4.0
Postcss-cssnext: 2.8.0

Here is my current webpack config: 

```
module.exports = {
...
module: {
    rules: [
      ...
      {
        test: /\.css$/,
        loaders: [
          'style-loader',
          {
            loader: 'css-loader',
            options: {
              importLoaders: 1,
              modules: true,
              localIdentName: '[path]___[name]__[local]___[hash:base64:5]',
            },
          },
          'postcss-loader',
        ],
      },
    ],
  },
```

Then my postcss.config.js: 

```
module.exports = {
  plugins: [
    require('postcss-smart-import')({}),
    require('postcss-simple-vars')({}),
    // require('postcss-cssnext')({}),
    // required('precss)({}),
  ],
};
```

With both postcss-cssnext and precss commented out, this works perfectly fine. If I allow even one of these two plugins then all of my plugins stop working and I get errors from webpack: TypeError: [object Object] must be a function, did you require() it ?

**Comment by michael-ciniawsky** [comment]
👋 

The missing `'` is in the example I assume :)

```js
// required('precss // <= ')({}),
```
Could be a bug because those two a packages, there was are PR handling that better, but not released yet since I hadn't found the time to test it out, maybe @Konvensky can elaborate on this :)

```

**Comment by Skilgarriff** [comment]
@michael-ciniawsky Yes, sorry about that - I added precss in while working in the Github editor as it wasn't still in my code, but I had tried it earlier.

**Comment by michael-ciniawsky** [comment]
@Skilgarriff few minutes please, out of the gut you can try to hotfix at least `postcss-cssnext` with

```
require('cssnext')({}).plugins
```

But not sure if it works tbh :)

**Comment by Skilgarriff** [comment]
@michael-ciniawsky No worries! hahaha I'll try and report back

**Comment by ai** [comment]
@Skilgarriff what errors exactly? We need stacktrace too.

**Comment by michael-ciniawsky** [comment]
@ai @Skilgarriff  Still on it, will take a day longer, this is 👎, any new info on this, errors, detail config etc ? :)

**Comment by Skilgarriff** [comment]
```
TypeError: [object Object] must be a function, did you require() it ?
    at /Users/Sean/Sites/Sean Kilgarriff/www/node_modules/postcss-load-plugins/lib/plugins.js:23:17
    at Array.forEach (native)
    at plugins (/Users/Sean/Sites/Sean Kilgarriff/www/node_modules/postcss-load-plugins/lib/plugins.js:21:15)
    at /Users/Sean/Sites/Sean Kilgarriff/www/node_modules/postcss-load-config/index.js:67:18
```

@ai 
Here is a stack trace for when I include postcss-cssnext as a plugin. 

@michael-ciniawsky No worries, I appreciate you working on it! No new information - I have basically recreated all of the functionality that I needed from these two plugins with other smaller plugins and not one of them has throw an error yet. Only getting broken Postcss/errors when I include one of those two.

**Comment by YasserKaddour** [comment]
I have the same error with the same trace while trying to upgrade to webpack 2:

I am using:
    "webpack": "2.1.0-beta.27",
    "postcss-cssnext": "^2.8.0",
    "postcss-loader": "1.1.0",
    "precss": "^1.4.0",
    "doiuse": "^2.5.0",

My postcss.config.js
```
module.exports = {
    plugins: [
        require('precss'),
        require('postcss-cssnext')(),
        require('doiuse')({
            browsers: ['ie >= 11', '> 1%'],
            ignore: [
                'flexbox',
                'border-radius'
            ]
        })
    ]
}
```
When I remove the parenthesis for the options like this, 
```
plugins: [
        require('precss'),
        require('postcss-cssnext'),
        require('doiuse')
    ]
```
It work.

Thank you for all your awesome contribution.

**Comment by princed** [comment]
@michael-ciniawsky I have the same issue with array form of config, although object form works fine. 

I've searched for differences between them and it looks like that the bug could be hidden here: [postcss-load-plugins](https://github.com/michael-ciniawsky/postcss-load-plugins/blob/62f0087278a0b2865d003d5b962f52fcf2e4c25b/lib/plugins.js#L23). It won't work probably because `plugin` isn't really replaced by `plugin.postcss` in array. However, I haven't time to check it yet.

**Comment by YasserKaddour** [comment]
Thanks @michael-ciniawsky, indeed this work for me
```
module.exports = {
    plugins: {
        'precss': {},
        'postcss-cssnext': {},
        'doiuse': {
            browsers: ['ie >= 11', '> 1%'],
            ignore: [
                'flexbox',
                'border-radius'
            ]
        }
    }
}
```

**Comment by MoOx** [comment]
Any reason why now some plugin called with () are crashing postcss-loader?

https://gitter.im/MoOx/postcss-cssnext?at=582eccc42291180a7a6637d8

**Comment by princed** [comment]
Actually it work just fine with https://github.com/michael-ciniawsky/postcss-load-plugins/pull/17.
So all we need is to publish latest version `postcss-load-plugins` as patch or minor version

@michael-ciniawsky Could you please do it?

**Comment by michael-ciniawsky** [comment]
@Skilgarriff  @princed  @YasserKaddour  Sry for the delay you guys, I do it today/tomorrow with other small fixes already on the list , I'm always busy with social work, there is no 'doing later' 😛 . Use {Object} style in your ``postcss.config.js``` plugins section meanwhile.

**Comment by folmert** [comment]
Confirmed: postcss-load-plugins does the job. But I'm still a bit confused - I ended up with 3 very similar modules:

- postcss-load-plugins
- postcss-load-options
- postcss-load-config

Any reason now to keep last two?

**Comment by azat-io** [comment]
I have the same issue with `css-mqpacker`:

I'm using:

* **webpack** v2.2.0-rc.2
* **postcss-loader** v1.2.1
* **css-mqpacker** v5.0.1

Here is a part of my  `webpack.config.babel.js`:

```js
export default {
    entry: './components/App/index.jsx',
    output: {
        path: path.join(process.cwd(), './public'),
        filename: 'main.[hash].js',
    },

    plugins: [
        new webpack.LoaderOptionsPlugin({
            options: {
                context: __dirname,
                postcss: [
                    cssnext({
                        browsers: ['last 2 versions', 'IE > 10'],
                    }),
                    cssMqpacker(),
                ],
            },
        }),

        new ExtractTextPlugin({
            filename: 'style.[hash].css',
            disable: false,
            allChanks: true,
        }),
    ],

    module: {
        loaders: [{
            test: /\.css$/,
            loader: ExtractTextPlugin.extract({
                fallbackLoader: 'style-loader',
                loader: 'css-loader?modules&localIdentName=[local]--[hash:base64:5]!postcss-loader',
            }),
        }],
    },
}
```

And `postcss-cssnext` and other plugins works good. But not `css-mqpacker`. When I'm building app I see the following CSS code:

```css
.title--Njw1N {
    color: red
}
@media (max-width: 800px) {
    .title--Njw1N {
        color: darkred
    }
}
.title--13b4T {
    color: yellow
}
@media (max-width: 800px) {
    .title--13b4T {
        color: orange
    }
}
.title--FYL1K {
    color: green
}
@media (max-width: 800px) {
    .title--FYL1K {
        color: darkgreen
    }
}
```

Please see the example here: **https://github.com/azat-io/webpack2-css-modules-demo**

**Comment by andykenward** [comment]
@azat-io [`css-mqpacker`](https://github.com/hail2u/node-css-mqpacker) doesn't seem to support sorting separate `.css` files together. Is it suppose to?

It worked fine when I added the [example](https://github.com/hail2u/node-css-mqpacker#synopsis) to just one of your `.css` files.

Might you have to run it on the Webpack CSS output afterwards?

**Comment by azat-io** [comment]
@anandthakker Oh, you are right. It looks like I need to concatenate all files before using CSS Mqpacker

**Comment by michael-ciniawsky** [comment]
fixed in #161 via `postcss-load-config` update + watching config file for changes is supported now
